lighthouse-extension/app/src/lighthouse-ext-background.js

```diff
@@ -70,117 +47,103 @@ function updateBadgeUI(optUrl) {
 }
 
 /**
- * @param {!Object} options Lighthouse options.
- * @param {!Array<string>} categoryIDs Name values of categories to include.
- * @return {!Promise}
+ * @param {{flags: LH.Flags}} options Lighthouse options.
+ * @param {Array<string>} categoryIDs Name values of categories to include.
+ * @return {Promise<LH.RunnerResult|void>}
  */
-window.runLighthouseInExtension = function(options, categoryIDs) {
+async function runLighthouseInExtension(options, categoryIDs) {
   // Default to 'info' logging level.
   log.setLevel('info');
   const connection = new ExtensionProtocol();
   options.flags = Object.assign({}, options.flags, {output: 'html'});
 
-  // return enableOtherChromeExtensions(false)
-  // .then(_ => connection.getCurrentTabURL())
-  return connection.getCurrentTabURL()
-    .then(url => window.runLighthouseForConnection(connection, url, options,
-      categoryIDs, updateBadgeUI))
-    .then(runnerResult => {
-      // return enableOtherChromeExtensions(true).then(_ => {
-      const blobURL = window.createReportPageAsBlob(runnerResult, 'extension');
-      return new Promise(resolve => chrome.windows.create({url: blobURL}, resolve));
-      // });
-    }).catch(err => {
-      // return enableOtherChromeExtensions(true).then(_ => {
-      throw err;
-      // });
-    });
-};
+  const url = await connection.getCurrentTabURL();
+  const runnerResult = await background.runLighthouseForConnection(connection, url, options,
+    categoryIDs, updateBadgeUI);
+  if (!runnerResult) {
+    // For now, should always be a runnerResult as the extension can't do `gatherMode`
+    throw new Error('no runnerResult generated by Lighthouse');
+  }
+
+  const blobURL = createReportPageAsBlob(runnerResult);
+  await new Promise(resolve => chrome.windows.create({url: blobURL}, resolve));
+}
 
 /**
  * Run lighthouse for connection and provide similar results as in CLI.
- * @param {!Connection} connection
+ * @param {Connection} connection
  * @param {string} url
- * @param {!Object} options Lighthouse options.
+ * @param {{flags: LH.Flags} & {outputFormat: string, logAssets: boolean}} options Lighthouse options.
           Specify outputFormat to change the output format.
- * @param {!Array<string>} categoryIDs Name values of categories to include.
- * @return {!Promise}
+ * @param {Array<string>} categoryIDs Name values of categories to include.
+ * @return {Promise<string|Array<string>|void>}
  */
-window.runLighthouseAsInCLI = function(connection, url, options, categoryIDs) {
+async function runLighthouseAsInCLI(connection, url, options, categoryIDs) {
   log.setLevel('info');
-  const startTime = Date.now();
   options.flags = Object.assign({}, options.flags, {output: options.outputFormat});
 
-  return window.runLighthouseForConnection(connection, url, options, categoryIDs)
-    .then(results => {
-      const endTime = Date.now();
-      results.timing = {total: endTime - startTime};
```

Comment:
currently LR continues to expect a .timing.total prop.

Reply:
> currently LR continues to expect a .timing.total prop.

this is [added in runner now](https://github.com/GoogleChrome/lighthouse/blob/c5743e9a376b709d4a0e74d8cec7f02095a8a573/lighthouse-core/runner.js#L135), so doesn't need to be done here